R/rocRAND/common.jl

```diff
@@ -0,0 +1,101 @@
+const ROCM_GIT = "https://github.com/ROCmSoftwarePlatform/rocRAND/"
+const NAME = "rocRAND"
+
+const GIT_TAGS = Dict(
+    v"4.2.0" => "15725c89e9cc9cc76bd30415fd2c0c5b354078831394ab8b23fe6633497b92c8",
+    v"4.5.2" => "1523997a21437c3b74d47a319d81f8cc44b8e96ec5174004944f2fb4629900db",
+)
+
+const ROCM_PLATFORMS = [
+    Platform("x86_64", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+    Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx11"),
+]
+const PRODUCTS = [
+    LibraryProduct(["librocrand"], :librocrand, ["rocrand/lib"]),
+    LibraryProduct(["libhiprand"], :libhiprand, ["hiprand/lib"]),
+]
+
+const BUILDSCRIPT = raw"""
+cd ${WORKSPACE}/srcdir/rocRAND*/
+mkdir build
+
+export ROCM_PATH=${prefix}
+export HIP_PATH=${prefix}/hip
+
+export HIP_PLATFORM=amd
+export HIP_RUNTIME=rocclr
+export HIP_COMPILER=clang
+export HSA_PATH=${prefix}
+# export HIP_ROCCLR_HOME=${prefix}/lib
+export HIP_CLANG_PATH=${prefix}/llvm/bin
+
+# Other HIPCC env variables.
+export HIPCC_VERBOSE=1
+export HIP_LIB_PATH=${prefix}/hip/lib
+export DEVICE_LIB_PATH=${prefix}/amdgcn/bitcode
+export HIP_CLANG_HCC_COMPAT_MODE=1
+
+# BB compile HIPCC flags:
+BB_COMPILE_BASE_DIR=/opt/${target}/${target}
+BB_COMPILE_CPP_DIR=${BB_COMPILE_BASE_DIR}/include/c++/*
+OMP_DIR=/opt/${target}/lib/gcc/${target}/*/include
+BB_COMPILE_FLAGS=" -isystem ${OMP_DIR} -isystem ${BB_COMPILE_CPP_DIR} -isystem ${BB_COMPILE_CPP_DIR}/${target} --sysroot=${BB_COMPILE_BASE_DIR}/sys-root"
+
+# BB link HIPCC flags:
+BB_LINK_GCC_DIR=/opt/${target}/lib/gcc/${target}/*
+BB_LINK_FLAGS=" --sysroot=${BB_COMPILE_BASE_DIR}/sys-root -B ${BB_LINK_GCC_DIR} -L ${BB_LINK_GCC_DIR}  -L ${BB_COMPILE_BASE_DIR}/lib64 -L ${prefix}/lib"
+
+# Set compile & link flags for hipcc.
+export HIPCC_COMPILE_FLAGS_APPEND=$BB_COMPILE_FLAGS
+export HIPCC_LINK_FLAGS_APPEND=$BB_LINK_FLAGS
+
+export PATH="${prefix}/hip/bin:${prefix}/llvm/bin:${PATH}"
+export LD_LIBRARY_PATH="${prefix}/lib:${prefix}/llvm/lib:${LD_LIBRARY_PATH}"
+
+# NOTE
+# Looking at hcc-cmd, it is clear that it is omitting 'hip/include' directory.
+# Therefore we symlink to other directory that it looks at.
+mkdir ${prefix}/lib/include
+ln -s ${prefix}/hip/include/* ${prefix}/lib/include
+
+# NOTE
+# This is needed to avoid errors with zipping files older than 1980.
+# See: https://github.com/pypa/wheel/issues/418
+unset SOURCE_DATE_EPOCH
+pip install -U pip wheel setuptools
+
+amdgpu_targets="gfx900"
```

Comment:
Can we expand this to include all GPUs supported by rocRAND?

Reply:
Done!